roofit/roofitcore/inc/RooNumber.h

```diff
@@ -38,6 +39,9 @@ class RooNumber {
    /// Get the absolute epsilon that is used by range checks in RooFit,
    /// e.g., in RooAbsRealLValue::inRange().
    inline static double rangeEpsAbs() { return staticRangeEpsAbs(); }
+   /// @brief  Returns an std::to_string compatible number (i.e. rounds infinities back to the nearest representable
```

Comment:
We usually put the documentation to function definition, not declaration. Can you move this to the `.cxx` file?

Also it would be good to explain why the function represents infinity with finite numbers, before this is forgotten. After all, `std::to_string(std::numeric_limits<double>::infinity())` works on most platforms (except for 32 bit if I remember correctly from the past CI reports).

And it's not only because of that! One could also consider returning the actual code strings for infinity like this:
```
   if (sign)
      return sign == 1 ? "std::numeric_limits<double>::infinity()" : "-std::numeric_limits<double>::infinity()";
```

But this doesn't work either because Clad can't deal with this and the derivatives are wrong! (they are infinite, even though `TMath::Erfc(infinity)` is finite and should be no problem.

So I think there is the following action to be taken here:
1. Note in the doc of `toString()` that this function is originally intended for generating differentiable code from RooFit models
2.  Explain that `std::to_string()` can't be used for infinity because it's not supported on all platforms, e.g. not on 32 bit Debian
3. Explain that also returning a string that encodes infinity in C++ (i.e. `"std::numeric_limits<double>::infinity()"`) would not work, because `TMath::Erfc(infinity)` can't be handled correctly Clad

4. Open a GitHub issue in the Clad or ROOT repo about this problem. I'm not sure if the problem here is Clad, or the derivative implementation of `TMath::Erfc()` in ROOT.

Point 4 is obviously disconnected from this PR, but I think it's still important because the lessons we learned here should now get forgotten.